traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/ds/DeliveryService.java

```diff
@@ -147,6 +148,7 @@ public DeliveryService(final String id, final JsonNode dsJo) throws JsonUtilsExc
 		}
 		sslEnabled = JsonUtils.optBoolean(dsJo, "sslEnabled");
 		this.anonymousIpEnabled = JsonUtils.optBoolean(dsJo, "anonymousBlockingEnabled");
+		this.consistentHashRegex = JsonUtils.optString(dsJo, "consistentHashRegex");
```

Comment:
I'm thinking we should probably compile the regex here so that it doesn't need to be compiled on every request. It's probably negligible in terms of the total amount of time that a single request makes, but it's just unnecessary work to do on every request that comes in for this DS.

Reply:
So it's pretty convenient to be able to pass a string around as the regex and let the `buildPatternBasedHashString(..)` method do the regex compilation and try/catch handling, especially for the code paths that don't use a delivery service (unit tests and API endpoints). Do you think that the benefit of your suggested approach outweighs the convenience of what is currently in place?

Reply:
I don't feel too strongly about it. It's hard to make any judgements about performance without doing any benchmarking or load testing, but just timing a simple regex compilation locally took between 0.05 - 0.1 ms. That seems pretty negligible to me, but if we start caring about fractions of a millisecond in `ttms` this is something we could optimize later down the line.